indico/modules/rb_new/util.py

```diff
@@ -638,19 +641,43 @@ def get_room_blockings(start_date=None, end_date=None, created_by=None, in_rooms
     return query.all()
 
 
-def create_blocking(rooms, start_date, end_date, reason, allowed_principals):
-    blocking = Blocking()
-    blocking.start_date = start_date
-    blocking.end_date = end_date
-    blocking.reason = reason
+def _group_id_or_name(principal):
+    provider = principal['provider']
+    if provider is None or provider == 'indico':
+        return principal['id']
+    else:
+        return principal['name']
+
+
+@no_autoflush
+def _populate_blocking(blocking, room_ids, reason, allowed_principals, start_date=None, end_date=None):
+    blocking.blocked_rooms = [BlockedRoom(room_id=room.id) for room in Room.query.filter(Room.id.in_(room_ids))]
     blocking.created_by_user = session.user
-    blocking.allowed = allowed_principals
-    blocking.blocked_rooms = [BlockedRoom(room_id=room.id) for room in rooms]
+    blocking.reason = reason
+    blocking.allowed = [GroupProxy(_group_id_or_name(pr), provider=pr['provider'])
+                        if pr.get('is_group')
+                        else User.get_one(pr['id'])
+                        for pr in allowed_principals]
+
+    if start_date:
```

Comment:
I would handle those in `create_blocking` (`blocking.start_date = blocking_data.pop('start_date')`)